test/functional/1-telemetry.js

```diff
@@ -0,0 +1,173 @@
+/* eslint-env node, mocha */
+
+// for unhandled promise rejection debugging
+process.on("unhandledRejection", r => console.error(r)); // eslint-disable-line no-console
+
+const {assert} = require("chai");
+const utils = require("./utils");
+const firefox = require("selenium-webdriver/firefox");
+const Context = firefox.Context;
+
+describe("telemetry", function() {
+  // This gives Firefox time to start, and us a bit longer during some of the tests.
+  this.timeout(15000);
+
+  let driver;
+  let studyPings;
+
+  // runs ONCE
+  before(async () => {
+    driver = await utils.setupWebdriver.promiseSetupDriver(
+      utils.FIREFOX_PREFERENCES,
+    );
+    await utils.setupWebdriver.installAddon(driver);
+  });
+
+  after(() => {
+    driver.quit();
+  });
+
+  function checkTelemetryPayload() {
+    it("has recorded one ping", async () => {
+      assert(studyPings.length == 1, "one shield telemetry ping");
+    });
+
+    it("correctly records etld as a hash", async () => {
+      let ping = studyPings[0];
+      let attributes = ping.payload.data.attributes;
+      assert.exists(attributes.etld, "etld exists");
+      assert.notInclude(attributes.etld, "itisatrap", "etld does not contain the domain");
+    });
+
+    it("correctly records page reloaded", async () => {
```

Comment:
I might change this assertion name, since it seems to imply that the page is being reloaded and recorded, but this is checking that it has not changed. I don't see any tests that actually are making a reload happen.

Reply:
Yeah, I was thinking I would test the reload scenario in a third test file which also tests the reload survey, but thinking about it I might as well test it here as well.